src/compile/data.ts

```diff
@@ -462,6 +447,23 @@ export namespace dates {
   }
 }
 
+// We need to add a rank transform so that we can use the rank value as
+// input for color ramp's linear scale.
+export function rankTransform(dataTable, model: Model) {
+  if (model.has(COLOR) && model.fieldDef(COLOR).type === ORDINAL) {
+    dataTable.transform = dataTable.transform.concat([{
```

Comment:
I actually prefer not causing side effect.   
(And yes, I also don't like how I wrote `filterNonPositiveForLog` earlier)